src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3982,14 +3988,176 @@ bool   os::win32::_is_windows_server         = false;
+  DWORD job_object_information_length = 0;
+
+  if (!QueryInformationJobObject(nullptr, JobObjectGroupInformationEx, nullptr, 0, &job_object_information_length)) {
+    DWORD last_error = GetLastError();
+    if (last_error == ERROR_INSUFFICIENT_BUFFER) {
+      DWORD group_count = job_object_information_length / sizeof(GROUP_AFFINITY);
+
+      job_object_information = os::malloc(job_object_information_length, mtInternal);
+      if (job_object_information != nullptr) {
+          if (QueryInformationJobObject(nullptr, JobObjectGroupInformationEx, job_object_information, job_object_information_length, &job_object_information_length)) {
+            assert(group_count == job_object_information_length / sizeof(GROUP_AFFINITY), "Unexpected group count");
+
+            for (DWORD i = 0; i < group_count; i++) {
+              KAFFINITY group_affinity = ((GROUP_AFFINITY*)job_object_information)[i].Mask;
+              processors += population_count(group_affinity);
+            }
+
+            assert(processors > 0, "Must find at least 1 logical processor");
+          } else {
+            warning("QueryInformationJobObject() failed: GetLastError->%ld.", GetLastError());
+          }
+
+          os::free(job_object_information);
+      } else {
+          warning("os::malloc() failed to allocate %ld bytes for QueryInformationJobObject", job_object_information_length);
+      }
+    }
+  }
+
+  return processors;
+}
+
+DWORD os::win32::system_logical_processor_count() {
+  DWORD logical_processors = 0;
+  LOGICAL_PROCESSOR_RELATIONSHIP relationship_type = RelationGroup;
+  PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX system_logical_processor_info = nullptr;
+  DWORD returned_length = 0;
+
+  // https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getlogicalprocessorinformationex
+  if (!GetLogicalProcessorInformationEx(relationship_type, nullptr, &returned_length)) {
+    DWORD last_error = GetLastError();
+
+    if (last_error == ERROR_INSUFFICIENT_BUFFER) {
+      system_logical_processor_info = (PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX)os::malloc(returned_length, mtInternal);
+
+      if (nullptr == system_logical_processor_info) {
+        warning("os::malloc() failed to allocate %ld bytes for GetLogicalProcessorInformationEx buffer", returned_length);
+      } else if (!GetLogicalProcessorInformationEx(relationship_type, system_logical_processor_info, &returned_length)) {
+        warning("GetLogicalProcessorInformationEx() failed: GetLastError->%ld.", GetLastError());
+      } else {
+        DWORD processor_groups = system_logical_processor_info->Group.ActiveGroupCount;
+
+        for (DWORD i = 0; i < processor_groups; i++) {
+          PROCESSOR_GROUP_INFO group_info = system_logical_processor_info->Group.GroupInfo[i];
+          logical_processors += group_info.ActiveProcessorCount;
+        }
+
+        assert(logical_processors > 0, "Must find at least 1 logical processor");
```

Comment:
Should this be a warning?  In release the assert won't trigger and you won't know that this was 0 and that in the calling method it fallsback to the other method for getting processor count

Reply:
> Should this be a warning? In release the assert won't trigger and you won't know that this was 0 and that in the calling method it fallsback to the other method for getting processor count

I've added a warning before the assert